libraries/GCS_MAVLink/GCS.cpp

```diff
@@ -155,13 +156,14 @@ void GCS::update_sensor_status_flags()
     }
 
     const AP_Logger &logger = AP::logger();
-    if (logger.logging_present()) {  // primary logging only (usually File)
+    const AP_GPS &gps = AP::gps();
+    if (logger.logging_present() || gps.drivers_should_log()) {  // primary logging only (usually File)
         control_sensors_present |= MAV_SYS_STATUS_LOGGING;
     }
-    if (logger.logging_enabled()) {
+    if (logger.logging_enabled() || gps.drivers_should_log()) {
         control_sensors_enabled |= MAV_SYS_STATUS_LOGGING;
     }
-    if (!logger.logging_failed()) {
+    if (!logger.logging_failed() && gps.logging_healthy()) {
```

Comment:
The methods were named quite deliberately for their intended purpose.  Logging isn't considered failed if we are not intending to log.  Asserting logging is healthy if you're not intending to right isn't possible, thus the "negative" aspect of the function.

Reply:
Look at what the logging_healthy() function does, if you aren't intending to log then it's considered healthy.